src/credentials/sso/ssoAccessTokenProvider.ts

```diff
@@ -4,93 +4,103 @@
+    ): Promise<SsoAccessToken> {
+        // Calculate the device code expirtation in milliseconds
+        const deviceCodeExpiration = Date.now() + authorization.expiresIn! * MILLISECONDS_PER_SECOND
 
-        //temporary logging to get URL
-        const logger = getLogger()
-        logger.info(
+        getLogger().info(
             `To complete authentication for this SSO account, please continue to this SSO portal:${authorization.verificationUriComplete}`
         )
 
         // The retry interval converted to milliseconds
-        let retryInterval = authorization.interval * 1000
+        let retryInterval =
+            authorization.interval! > 0 ? authorization.interval! * MILLISECONDS_PER_SECOND : BACKOFF_TIME
 
         const createTokenParams = {
             clientId: registration.clientId,
             clientSecret: registration.clientSecret,
             grantType: GRANT_TYPE,
-            deviceCode: authorization.deviceCode,
+            deviceCode: authorization.deviceCode!,
         }
 
         while (true) {
             try {
                 const tokenResponse = await this.ssoOidcClient.createToken(createTokenParams).promise()
-                const accessToken: AccessToken = {
+                const accessToken: SsoAccessToken = {
                     startUrl: this.ssoUrl,
                     region: this.ssoRegion,
                     accessToken: tokenResponse.accessToken!,
-                    expiresAt: new Date(Date.now() + tokenResponse.expiresIn! * 1000).toISOString(),
+                    expiresAt: new Date(Date.now() + tokenResponse.expiresIn! * MILLISECONDS_PER_SECOND).toISOString(),
                 }
                 return accessToken
             } catch (err) {
                 if (err.code === 'SlowDownException') {
-                    retryInterval += 5000
+                    retryInterval += BACKOFF_TIME
                 } else if (err.code === 'AuthorizationPendingException') {
-                    if (Date.now() + retryInterval > deviceCodeExpiration) {
-                        throw Error(
-                            `Device code has expired while polling for SSO token, login flow must be re-initiated.`
-                        )
-                    }
-                    // else, wait the interval and try again
+                    // do nothing, wait the interval and try again
                 } else if (err.code === 'ExpiredTokenException') {
                     throw Error(`Device code has expired while polling for SSO token, login flow must be re-initiated.`)
+                } else {
+                    getLogger().info(err)
+                    throw err
                 }
             }
+            if (Date.now() + retryInterval > deviceCodeExpiration) {
+                throw Error(`Device code has expired while polling for SSO token, login flow must be re-initiated.`)
+            }
             setTimeout(() => {}, retryInterval)
+            retryInterval *= 2
```

Comment:
Please check the spec, I seem to recall that the interval is to double if a timeout error occurs, else it remains as-is.
(AuthorizationPendingException is not a reason to double the interval)